packages/core/src/utils/spanUtils.ts

```diff
@@ -26,3 +27,34 @@ export function spanToTraceContext(span: Span): TraceContext {
 export function spanToTraceHeader(span: Span): string {
   return generateSentryTraceHeader(span.traceId, span.spanId, span.sampled);
 }
+
+/**
+ * Convert a span time input intp a timestamp in seconds.
+ */
+export function spanTimeInputToSeconds(input: SpanTimeInput | undefined): number {
+  if (typeof input === 'number') {
+    return ensureTimestampInSeconds(input);
+  }
+
+  if (Array.isArray(input)) {
+    /**
+     * Defines High-Resolution Time.
+     *
+     * The first number, HrTime[0], is UNIX Epoch time in seconds since 00:00:00 UTC on 1 January 1970.
+     * The second number, HrTime[1], represents the partial second elapsed since Unix Epoch time represented by first number in nanoseconds.
+     * For example, 2021-01-01T12:30:10.150Z in UNIX Epoch time in milliseconds is represented as 1609504210150.
+     * The first number is calculated by converting and truncating the Epoch time in milliseconds to seconds:
+     * HrTime[0] = Math.trunc(1609504210150 / 1000) = 1609504210.
+     * The second number is calculated by converting the digits after the decimal point of the subtraction, (1609504210150 / 1000) - HrTime[0], to nanoseconds:
+     * HrTime[1] = Number((1609504210.150 - HrTime[0]).toFixed(9)) * 1e9 = 150000000.
+     * This is represented in HrTime format as [1609504210, 150000000].
+     */
+    return input[0] + input[1] / 1e9;
```

Comment:
If this is vendored we need to add license

Reply:
it is only inspired, not copied - only the comment I took verbatim 🤔 but we can also move this to it's own file, then we can still add the licence even if it is adapted from there?

Reply:
with these kind of things, better be safe than sorry - let's move to it's own file.

Reply:
done!